go.mod

```diff
@@ -1,11 +1,15 @@
 module github.com/cockroachdb/swiss
 
-go 1.20
+go 1.22.0
```

Comment:
Did you mean to change this? I don't have a strong need to support older versions of Go, but want to make sure this was intentional. If it was, we'll need to adjust `.github/workflows/ci.yaml` to not test on go1.20 or go1.21.

Reply:
I used a few range-over-int loops in go-perfevent, which bumped this version. I took those out so now go-perfevent uses `go 1.20` and this can stay as-is.

Reply:
```
Error: ../../../go/pkg/mod/github.com/aclements/go-perfevent@v0.0.0-20240228224405-280aa97c1800/perf/counter.go:235:15: undefined: binary.NativeEndian
Error: ../../../go/pkg/mod/github.com/aclements/go-perfevent@v0.0.0-20240228224405-280aa97c1800/perf/counter.go:240:24: undefined: binary.NativeEndian
Error: ../../../go/pkg/mod/github.com/aclements/go-perfevent@v0.0.0-20240228224405-280aa97c1800/perf/counter.go:241:24: undefined: binary.NativeEndian
Error: ../../../go/pkg/mod/github.com/aclements/go-perfevent@v0.0.0-20240228224405-280aa97c1800/perf/counter.go:245:27: undefined: binary.NativeEndian
```

I'm fine with bumping the minimum version go1.21. That would also allow the use of `min` and `max` which I explicitly avoided so far.

Reply:
Ugh. That's what I get for not actually testing with Go 1.20. Pushed an update to go-perfevent to require Go 1.21, and a new patch here to update the dependency and Go version.

Reply:
You're going to have to delete go1.20 from the [.github/workflows/ci.yaml](https://github.com/cockroachdb/swiss/blob/main/.github/workflows/ci.yaml#L17). Should be a one line change.

Reply:
Done. (Sorry, I'm not very used to working with GitHub workflows.)